feg/gateway/services/session_proxy/credit_control/gy/gy_client.go

```diff
@@ -316,9 +317,15 @@ func getServiceInfoAvp(server *diameter.DiameterServerConfig, request *CreditCon
 		psInfoGrp.AddAVP(diam.NewAVP(avp.TGPPSGSNMCCMNC, avp.Vbit, diameter.Vendor3GPP, datatype.UTF8String(request.PlmnID)))
 		psInfoGrp.AddAVP(diam.NewAVP(avp.TGPPGGSNMCCMNC, avp.Vbit, diameter.Vendor3GPP, datatype.UTF8String(request.PlmnID)))
 	}
+
 	apn := datatype.UTF8String(request.Apn)
-	if len(gyGlobalConfig.OCSOverwriteApn) > 0 {
-		apn = datatype.UTF8String(gyGlobalConfig.OCSOverwriteApn)
+	if gyGlobalConfig != nil {
```

Comment:
This should also be a function.

See that it may look silly I am asking for functions, but functions are important to avoid the code to get messy. Adding a function with a good name may help the reader to avoid part of the code that may distract them. Even you, after having wrote this, it will be easier to read if you just see the name of a function instead of 7 lines of code.

For example here you can wrap this on a function named `getAPNfromConfig` and you will end up having
```
	if len(getAPNfromConfig(....)) > 0 {
		psInfoGrp.AddAVP(diam.NewAVP(avp.CalledStationID, avp.Mbit, 0, apn))			x
```

Reply:
Sure.